### PR TITLE
AWS Proxy in UCP

### PR DESCRIPTION
**Pull request by vinayada1**
# Description

Refactored branch: https://github.com/project-radius/radius/tree/rynowak/aws-plane for incorporating AWS Proxy changes in UCP

#3663 

Fixes: #3663

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Adds necessary unit tests for change
* [ ] Adds necessary E2E tests for change
* [ ] Unit tests passing
* [ ] Extended the documentation / Created issue for it

### CODE REVIEW

**Review by rynowak — COMMENTED** [review]
This is a pretty good start, it all needs tests 😆

**Review by rynowak — COMMENTED** [review]
Commenting here for posterity 😆 

I'm going to be out for 3 days, so don't block this PR on me. All of the original work from my spike is represented here, the only things missing are tests and any refactors we want to do. I'll leave that up to the rest of to team B 👍

**Review by jkotalik — APPROVED** [review]
I don't think anything here needs to block continued iteration on this in follow up PRs.

**Pull request merged**
